wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -348,27 +348,26 @@ public boolean getStickButtonPressed(final int stick, final int button) {
     if (stick < 0 || stick >= kJoystickPorts) {
       throw new IllegalArgumentException("Joystick index is out of range, should be 0-3");
     }
-    boolean error = false;
-    boolean retVal = false;
-    synchronized (m_cacheDataMutex) {
+    m_cacheDataMutex.lock();
+    try {
       if (button > m_joystickButtons[stick].m_count) {
-        error = true;
-        retVal = false;
-      } else {
-        // If button was pressed, clear flag and return true
-        if ((m_joystickButtonsPressed[stick] & 1 << (button - 1)) != 0) {
-          m_joystickButtonsPressed[stick] &= ~(1 << (button - 1));
-          retVal = true;
-        } else {
-          retVal = false;
-        }
+        // Unlock early so error printing isn't locked.
+        m_cacheDataMutex.unlock();
+        reportJoystickUnpluggedWarning("Joystick Button " + button + " on port " + stick
+            + " not available, check if controller is plugged in");
+      }
```

Comment:
Shouldn’t it immediately return false here?

Reply:
I wondered this as well briefly. If we want to preserve the current behaviour then yes it definitely should.

Doing it this way does allow the possibility of catching an input as a joystick is unplugged during a match, but the potential footgun of an input latching from a disabled-to-enabled transition (I'm assuming most teams don't call this method during disabled) could be dangerous, so I'm more than happy to change this if people agree.

Reply:
Thinking about it some more, the current behaviour currently also means that an input from a unplugged joystick will also latch when the joystick is plugged back in, I think. Not sure whether that's something we want to fix.

Reply:
Calling this method during disabled is pretty useful (as it can be used to e.g. select autonomous modes).  I'd recommed fixing this to preserve the current behavior.